addons/account/models/account_payment.py

```diff
@@ -211,6 +211,8 @@ def _onchange_currency(self):
         self.amount = abs(self._compute_payment_amount())
 
         # Set by default the first liquidity journal having this currency if exists.
+        if not self.currency_id or self.currency_id == self.journal_id.currency_id:
+            return
         journal = self.env['account.journal'].search(
             [('type', 'in', ('bank', 'cash')), ('currency_id', '=', self.currency_id.id)], limit=1)
         if journal:
```

Comment:
here's my proposal
``` diff
-        journal = self.env['account.journal'].search(
-            [('type', 'in', ('bank', 'cash')), ('currency_id', '=', self.currency_id.id)], limit=1)
-        if journal:
+        if not self.journal:
+            journal = self.env['account.journal'].search(
+                [('type', 'in', ('bank', 'cash')), ('currency_id', '=', self.currency_id.id)], limit=1)
             return {'value': {'journal_id': journal.id}}
```

Reply:
@qdp-odoo   
Done, I left the onchange without effect when the currency is already set.

I also edited commit/PR's description to reflect this behavior.